pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -226,13 +226,33 @@ def boxplot(data, column=None, by=None, ax=None, fontsize=None,
     def _get_colors():
         #  num_colors=3 is required as method maybe_color_bp takes the colors
         #  in positions 0 and 2.
-        return _get_standard_colors(color=kwds.get('color'), num_colors=3)
+        #  if colors not provided, use color 2 for medians and 0 for all else
+        result = _get_standard_colors(num_colors=3)
+        result = np.take(result, [0, 0, 2, 0])
+
+        colors = kwds.pop('color', None)
+        if colors:
+            if isinstance(colors, dict):
+                valid_keys = ['boxes', 'whiskers', 'medians', 'caps']
+                key_to_index = dict(zip(valid_keys, range(4)))
+                for key, value in colors.items():
```

Comment:
can you add a bit more comments on what you are trying to achieve here?

maybe best to just put a more full doc-string at the top